src/coreclr/vm/gcenv.ee.cpp

```diff
@@ -1651,6 +1651,19 @@ void GCToEEInterface::AnalyzeSurvivorsFinished(size_t gcIndex, int condemnedGene
             s_forcedGCInProgress = false;
             reportGenerationBounds();
             FireEtwGenAwareEnd((int)gcIndex, GetClrInstanceId());
+            if (gcGenAnalysisDump)
+            {
+                EX_TRY
+                {
+#ifdef HOST_WIN32
+                    GenerateCrashDump (GENAWARE_DUMP_FILE_NAME, 2, false);
+#else
+                    PAL_GenerateCoreDump (GENAWARE_DUMP_FILE_NAME, 2, false);
```

Comment:
do we need to still capture a trace in this case seeing that we are already capturing a dump? you can load a dump in perfview and the genaware analysis would show up..

Reply:
I made collecting a trace optional. There are a few reasons behind the design choice I made:

1. `COMPlus_GCGenAnalysisTrace` has 1 as a default - this is to maintain backward compatibility - users are expected to get a trace when both `GCGenAnalysisGen` and `GCGenAnalysisBytes` are set.

2. I make it possible to obtain both a trace and a dump at the same time. This is for the case of a Linux dump, where we might not be able to use the GenAware views in PerfView.